basilica/basilica/__init__.py

```diff
@@ -41,6 +43,9 @@ def __init__(self, auth_key, server='https://api.basilica.ai',
         self.adapter = HTTPAdapter(max_retries=self.retry)
         self.session.mount('http://', self.adapter)
         self.session.mount('https://', self.adapter)
+        self.write_queue = Queue.Queue()
+        self.read_queue = Queue.Queue()
+        self.thread_running = False
```

Comment:
I don't think we can have these queues on a per-connection level, because the same connection might be used for multiple calls to `embed`, and if this were done on two threads then the responses from the different `embed` calls would be mixed together.  I think we should construct this queue right before we spawn the worker thread, and pass the queues to the worker thread.

Reply:
We should probably have a test where `embed` is called on the same connection while an existing `embed` call is ongoing.